packages/views.client.unstable/src/client.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type {
+  HostMessage,
+  ParameterConfig,
+  ViewEmitEventMessage,
+  ViewMessage,
+} from "@osdk/views-api.unstable";
+import { META_TAG_HOST_ORIGIN } from "@osdk/views-api.unstable";
+
+type FoundryHostEventListener<CONFIG extends ParameterConfig> = (
+  event: MessageEvent<HostMessage<CONFIG>>,
+) => void;
+
+export interface IFoundryViewClient<CONFIG extends ParameterConfig> {
+  ready: () => void;
+  emit: (
+    message: Extract<ViewMessage<CONFIG>, ViewEmitEventMessage<CONFIG>>,
+  ) => void;
+}
+
+export class FoundryViewClient<CONFIG extends ParameterConfig>
```

Comment:
I have mixed feelings about exporting classes. I think they can severely limit what we are capable of doing and open us up to weird things like people trying to extend them when they should really be composing.

If we stick with a class, why would anyone want to use the `IFoundryView` type when they have to deal with a FoundryViewClient type to construct? What are we gaining?

If we go with function based `createFoundryViewClient()` then I would rename `IFoundryViewClient` to just `FoundryViewClient` and `FoundryViewClient` to `FoundryViewClientImpl`. 

I think I am biased towards exporting a function and never using `I` interfaces. I think it feels weird and don't really see it in any code in the wild. Even in this PR we only use it here once even though there are a TON of interfaces.

Reply:
`IFoundryView` mostly existed so i could grab the type to use on the context, but there are obviously better ways of doing that. i don't feel strongly one way or the other, so happy to switch to a function